main/src/ca/uwaterloo/flix/language/phase/unification/set/SetFormula.scala

```diff
@@ -872,4 +872,146 @@ object SetFormula {
     }
   }
 
+  /**
+    * The Successive Variable Elimination algorithm.
+    *
+    * Returns the most-general unifier of the equation `f ~ empty` where `fvs` is the free
+    * variables in `f`. If there is no unifier then [[NoSolutionException]] is thrown.
+    *
+    * Eliminates variables one-by-one from the given list `fvs` recursively.
+    *
+    * If the recursively building formula is ever larger than `recSizeThreshold` then
+    * [[ComplexException]] is thrown. If `recSizeThreshold` is non-positive then there is no
```

Comment:
Why do it like this? I would just prefer people can use Int.MaxValue or whatever

Reply:
that would mean that we always call `f.size` which is expensive. But you can still call it with `MaxValue` if you want to.